pkg/test/fakes/gce_metadata_server/Dockerfile

```diff
@@ -0,0 +1,4 @@
+FROM scratch
```

Comment:
How will we deploy this in the tests? Specifically wonder why it needs to be a standalone docker image vs part of echo or something?

Reply:
It could be bundled as a part of something else, I guess. My initial thought is that we only want this for a very few targeted tests, and so I didn't want to glom on to something unrelated.

I was planning on defining a separate component (much like the Stackdriver fake), that would be spun up. Then, we grab the IP and set that as an env var in the containers so that the bootstrap stuff looks it up instead of the default address.

Reply:
I am fine either way. One issue if its part of echo is that envoy may start before it does, which may introduce errors. Maybe that is desirable for fault injection, but I am pretty sure that will break things?

Reply:
Sent a PR to run it as a separate component to start: https://github.com/istio/istio/pull/25287.